packages/reports/addon/components/filter-values/dimension-select.js

```diff
@@ -49,9 +49,13 @@ export default Ember.Component.extend({
         dimensionName = get(this, 'dimensionName'),
         dimensionService = get(this, '_dimensionService');
 
-        // Only fetch dimensions if filter has values
+    // Only fetch dimensions if filter has values
     if (get(dimensionIds, 'length')) {
-      return dimensionService.find(dimensionName, {field: 'id', values: dimensionIds.join(',')});
+      return dimensionService.find(dimensionName, {field: 'id', values: dimensionIds.join(',')}).then(dims => dims.map(dimension => {
+        //adding isEqual to object so power select can find selected values accurately.
+        dimension.isEqual = function(other) { return this.id == other.id; };
```

Comment:
`(other) => this.id === other.id`

Reply:
should we add a test for the problem this fixes?

Reply:
I think arrow function will not work here due to `this` scope

Reply:
correct arrow function is not appropriate here.

Reply:
ok.. we want `this` to be in the dimension's scope.

Reply:
Can we move `isEqual` to the dimension model:

https://github.com/yahoo/navi/blob/master/packages/data/addon/models/bard-dimension.js#L9